### PR TITLE
Add custom creatures to the hero edit window.

### PR DESCRIPTION
**Pull request by mkristofik**

### CODE REVIEW

File: src/cpp/editor/hero_edit.cpp [hunk 1]
```diff
@@ -10,10 +11,24 @@ extern HeroExtra gEditedHeroExtra;
 
 namespace {
   bool shouldFillInArtifacts = true;
+  bool shouldFillInCreatures = true;
+
+  std::string Capitalize(const char *name) {
```
Comment (jkoppel) [inline src/cpp/editor/hero_edit.cpp]: I find it a bit weird to accept  char* but return a string. This is a file-private function so it doesn't really matter.

Comment (mkristofik) [reply src/cpp/editor/hero_edit.cpp]: `std::string` has undefined behavior if you pass a `nullptr` to the constructor.  I didn't trust what was in `gArmyNames` so I did it this way to be careful.

Comment (mkristofik) [reply src/cpp/editor/hero_edit.cpp]: And you told me in the past you preferred `std::string` anyway.

Comment (jkoppel) [reply src/cpp/editor/hero_edit.cpp]: Yeah; that's written in the Ironfist guidelines. I'd still generally expect a Capitalize function to take and return the same type.

**Pull request merged**